### PR TITLE
Update addon

### PR DESCRIPTION
**Pull request by shokmaster**
## Tasks

- [x] Add new CI
- [x] Update to Ember v3.8
- [x] Increase coverage and thresholds

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,50 +32,74 @@
...
-    "eslint-config-bbva": "^2.0.0",
-    "loader.js": "^4.2.3"
+    "ember-load-initializers": "^2.0.0",
+    "ember-maybe-import-regenerator": "^0.1.6",
+    "ember-qunit": "^4.4.1",
+    "ember-resolver": "^5.0.1",
+    "ember-sinon": "^3.1.0",
+    "ember-source": "~3.8.0",
+    "ember-source-channel-url": "^1.1.0",
+    "ember-try": "^1.1.0",
```
Comment (adrigzr) [inline package.json]: Remember to downgrade this version.

File: package.json [hunk 2]
```diff
@@ -28,50 +32,74 @@
...
+    "check-coverage": "istanbul check-coverage coverage/coverage-final.json --statements 80 --functions 90 --branches 65 --lines 80",
+    "commit": "git-cz",
+    "posttest": "npm run report-coverage && npm run check-coverage",
+    "report-coverage": "istanbul report --include=coverage/coverage-final.json text",
+    "semantic-release": "semantic-release"
   },
   "dependencies": {
-    "ember-cli-babel": "^6.6.0",
-    "ember-cli-htmlbars": "^2.0.3"
+    "ember-cli-babel": "^7.1.2"
```
Comment (adrigzr) [inline package.json]: This version can be upgraded.

File: testem.js [hunk 3]
```diff
@@ -1,8 +1,7 @@
-/* eslint-env node */
+/* eslint-disable no-process-env */
```
Comment (adrigzr) [inline testem.js]: This is not needed. Migrated scripts already updated with this change.

File: package.json (hunk 1 above)
Comment (shokmaster) [reply package.json]: done, mr hawk! :P

File: package.json (hunk 2 above)
Comment (shokmaster) [reply package.json]: upgraded to 7.5

File: testem.js (hunk 3 above)
Comment (shokmaster) [reply testem.js]: done

File: addon/utils/wait-for.js [hunk 4]
```diff
@@ -29,14 +30,14 @@ function runTimeout(func, wait) {
  * @method waitFor
  * @param  Function conditional
  * @param  Number wait
- * @return Ember.RSVP.Promise
+ * @return Promise
  * @public
  */
 export default function waitFor(conditional, wait = defaultWait, timeout = defaultTimeout) {
```
Comment (adrigzr) [inline addon/utils/wait-for.js]: I think we can remove this util with the equivalent [`waitUntil`](https://github.com/emberjs/ember-test-helpers/blob/master/addon-test-support/%40ember/test-helpers/wait-until.ts)

Comment (next-juanantoniogomez) [reply addon/utils/wait-for.js]: deleted

**Pull request merged**